source/common/signal/fatal_error_handler.cc

```diff
@@ -22,6 +27,26 @@ ABSL_CONST_INIT static absl::Mutex failure_mutex(absl::kConstInit);
 using FailureFunctionList = std::list<const FatalErrorHandlerInterface*>;
 ABSL_CONST_INIT std::atomic<FailureFunctionList*> fatal_error_handlers{nullptr};
 
+// Use an atomic operation since on fatal error we'll consume the
+// fatal_action_manager and don't want to have any locks as they aren't
+// async-signal-safe.
+ABSL_CONST_INIT std::atomic<FatalAction::FatalActionManager*> fatal_action_manager{nullptr};
+ABSL_CONST_INIT std::atomic<int64_t> failure_tid{-1};
+
+// Helper function to run fatal actions.
+void runFatalActions(const FatalAction::FatalActionPtrList& actions) {
+  FailureFunctionList* list = fatal_error_handlers.exchange(nullptr, std::memory_order_relaxed);
```

Comment:
Can you add more comments in this function about the atomic operations that we are doing and why we do them?